nifi-nar-bundles/nifi-stateless-processor-bundle/nifi-stateless-processor/src/main/java/org/apache/nifi/processors/stateless/ExecuteStateless.java

```diff
@@ -0,0 +1,1202 @@
+            createdSet = createOutputFlowFiles(optionalResult.get(), session, flowFile);
+        } catch (final IOException e) {
+            getLogger().error("Failed to write FlowFile contents that were output from Stateless Flow to the NiFi content repository for {}. Routing to failure.", flowFile, e);
+            if (flowFile != null) {
+                session.transfer(flowFile, REL_FAILURE);
+            }
+            return;
+        }
+
+        // Update any counters
+        adjustCounters(session, dataflow, null);
+
+        // If dataflow is yielded, yield this processor
+        final long yieldExpiration = dataflow.getSourceYieldExpiration();
+        if (yieldExpiration > 0) {
+            final long now = System.currentTimeMillis();
+            final long yieldMillis = yieldExpiration - now;
+            if (yieldMillis > 0) {
+                context.yield();
+            }
+        }
+
+        // Transfer the FlowFiles and asynchronously commit the session.
+        if (flowFile != null) {
+            session.transfer(flowFile, REL_ORIGINAL);
+        }
+        session.transfer(createdSet, REL_OUTPUT);
+        session.commitAsync(triggerResult::acknowledge);
+
+        if (flowFile == null) {
+            getLogger().info("Successfully triggered dataflow to run, producing {} output FlowFiles", createdSet.size());
+        } else {
+            getLogger().info("Successfully triggered dataflow to run against {}, producing {} output FlowFiles", flowFile, createdSet.size());
+        }
+
+        session.adjustCounter("Successful Invocations", 1, false);
+    }
+
+    private void surfaceBulletins(final BulletinRepository bulletinRepository, final long minBulletinId) {
+        // If there are any WARNING or ERROR bulletins, we want to log them for this processor. All of the log messages from the components
+        // themselves will already have been logged, but we want to surface any warn/error message as bulletins so we log them again for this processor.
+        final BulletinQuery bulletinQuery = new BulletinQuery.Builder()
+            .after(minBulletinId)
+            .build();
+
+        final List<Bulletin> bulletins = bulletinRepository.findBulletins(bulletinQuery);
+        for (final Bulletin bulletin : bulletins) {
+            try {
+                final LogLevel logLevel = LogLevel.valueOf(bulletin.getLevel());
+                if (logLevel == LogLevel.DEBUG || logLevel == LogLevel.INFO) {
+                    continue;
+                }
+
+                getLogger().log(logLevel, "{} {}[name={}, id={}] {}", bulletin.getTimestamp(), bulletin.getSourceType(), bulletin.getSourceName(), bulletin.getSourceName(), bulletin.getMessage());
+            } catch (final Exception ignored) {
+                // There should be no malformed bulletin, but just in case the bulletin level, etc. ends up being null, catch the exception and ignore it,
+                // since there's not much we can do about it.
```

Comment:
Is there any value in logging this exception condition as a debug or trace?

Reply:
Yeah, this should never happen. If it does, i'll actually log a warning.